make/Images.gmk

```diff
@@ -114,67 +114,55 @@ $(eval $(call SetupExecute, jlink_jre, \
 
 JLINK_JRE_TARGETS := $(jlink_jre)
 
-ifeq ($(BUILD_CDS_ARCHIVE), true)
+# Helper function for creating the CDS archives for the JDK and JRE
+#
+# Param1 - VM variant (e.g., server, client, zero, ...)
+# Param2 - _nocoops, or empty
+define CreateCDSArchive
+  $1_$2_DUMP_EXTRA_ARG := $(if $(filter _nocoops, $2),-XX:-UseCompressedOops,)
+  $1_$2_DUMP_TYPE      := $(if $(filter _nocoops, $2),-NOCOOPS,)
+
   ifeq ($(OPENJDK_TARGET_OS), windows)
-    CDS_ARCHIVE := bin/server/classes.jsa
-    CDS_NOCOOPS_ARCHIVE := bin/server/classes_nocoops.jsa
+    $1_$2_CDS_ARCHIVE := bin/$1/classes$2.jsa
   else
-    CDS_ARCHIVE := lib/server/classes.jsa
-    CDS_NOCOOPS_ARCHIVE := lib/server/classes_nocoops.jsa
+    $1_$2_CDS_ARCHIVE := lib/$1/classes$2.jsa
   endif
 
-  $(eval $(call SetupExecute, gen_cds_archive_jdk, \
-      WARN := Creating CDS archive for jdk image, \
-      DEPS := $(jlink_jdk), \
-      OUTPUT_FILE := $(JDK_IMAGE_DIR)/$(CDS_ARCHIVE), \
-      SUPPORT_DIR := $(SUPPORT_OUTPUTDIR)/images/jdk, \
-      COMMAND := $(FIXPATH) $(JDK_IMAGE_DIR)/bin/java -Xshare:dump \
-          -XX:SharedArchiveFile=$(JDK_IMAGE_DIR)/$(CDS_ARCHIVE) \
-          -Xmx128M -Xms128M $(LOG_INFO), \
+  $$(eval $$(call SetupExecute, $1_$2_gen_cds_archive_jdk, \
+      WARN := Creating CDS$$($1_$2_DUMP_TYPE) archive for jdk image for $1, \
+      DEPS := $$(jlink_jdk), \
+      OUTPUT_FILE := $$(JDK_IMAGE_DIR)/$$($1_$2_CDS_ARCHIVE), \
+      SUPPORT_DIR := $$(SUPPORT_OUTPUTDIR)/images/jdk, \
+      COMMAND := $$(FIXPATH) $$(JDK_IMAGE_DIR)/bin/java -Xshare:dump \
+          -XX:SharedArchiveFile=$$(JDK_IMAGE_DIR)/$$($1_$2_CDS_ARCHIVE) \
+          -$1 $$($1_$2_DUMP_EXTRA_ARG) -Xmx128M -Xms128M $$(LOG_INFO), \
   ))
 
-  JDK_TARGETS += $(gen_cds_archive_jdk)
+  JDK_TARGETS += $$($1_$2_gen_cds_archive_jdk)
 
-  $(eval $(call SetupExecute, gen_cds_archive_jre, \
-      WARN := Creating CDS archive for jre image, \
-      DEPS := $(jlink_jre), \
-      OUTPUT_FILE := $(JRE_IMAGE_DIR)/$(CDS_ARCHIVE), \
-      SUPPORT_DIR := $(SUPPORT_OUTPUTDIR)/images/jre, \
-      COMMAND := $(FIXPATH) $(JRE_IMAGE_DIR)/bin/java -Xshare:dump \
-          -XX:SharedArchiveFile=$(JRE_IMAGE_DIR)/$(CDS_ARCHIVE) \
-          -Xmx128M -Xms128M $(LOG_INFO), \
+  $$(eval $$(call SetupExecute, $1_$2_gen_cds_archive_jre, \
+      WARN := Creating CDS$$($1_$2_DUMP_TYPE) archive for jre image, \
```

Comment:
Perhaps include the variant name in the log message here, as is done above for the jdk image?